src/Schema/OracleSchemaManager.php

```diff
@@ -320,6 +320,109 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        if ($currentDatabase === null) {
+            throw new Exception('Invalid database name.');
+        }
+
+        $tableNames = $this->listTableNames();
+
+        $objectsByTable = [];
+
+        // Get all column definitions in one database call.
+        $sql = <<<'SQL'
+          SELECT C.*, D.COMMENTS AS COMMENTS
+            FROM ALL_TAB_COLUMNS C
+       LEFT JOIN ALL_COL_COMMENTS D ON D.OWNER = C.OWNER AND D.TABLE_NAME = C.TABLE_NAME AND
+                 D.COLUMN_NAME = C.COLUMN_NAME
+           WHERE C.OWNER =
+SQL;
+
+        $sql .= $this->_platform->quoteStringLiteral($currentDatabase);
```

Comment:
Now, instead of using string literals, we can and should use prepared statements.

Reply:
Done